tests/unit/ModelTypes.inc

```diff
@@ -0,0 +1,631 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
```

Comment:
Don't know where to put this, but really thanks a lot for this effort of testing stuff.

Reply:
Unfortunately I can't take credit for all this test code. I just adapted the model type test for Clift.